### PR TITLE
[5.0] Add type-hint to Manager constructor

### PR DESCRIPTION
**Pull request by martinbean**
Added `Application` type-hint to `Manager` constructor, so extending classes can be resolved by the service container.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Needs to be the application contract.

**Comment by martinbean** [comment]
@GrahamCampbell Could you have not left this open, so I could just push a new commit with the contract type-hint?

**Comment by GrahamCampbell** [comment]
Go for it. Please also update the relevant docblocks.

**Comment by martinbean** [comment]
@GrahamCampbell I already had. Second commit (after your first reply) has come through now you’re re-opened the PR. Docblocks were updated too.

File: src/Illuminate/Support/Manager.php [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <?php namespace Illuminate\Support;
 
 use Closure;
+use Illuminate\Contracts\Foundation\Application;
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Manager.php]: Please order these by length, not alphabetically.

And squash to one commit when you've fixed this.

**Comment by martinbean** [comment]
@GrahamCampbell Name-spaces ordered by length, and commits squashing down to one.

**Comment by ibrasho** [comment]
Should the type-hint be for `Illuminate\Contracts\Container\Container`?

**Comment by GrahamCampbell** [comment]
No.

**Comment by ibrasho** [comment]
> so extending classes can be resolved by the service container.

I thought the point was to be able to use the service container/locater.

**Comment by martinbean** [comment]
@ibrasho `Application` extends `Container`.

**Comment by taylorotwell** [comment]
Breaking change.

**Comment by martinbean** [comment]
@taylorotwell How so? All I did was add a type hint.

**Comment by crynobone** [comment]
Someone could use a manager outside of Laravel, which doesn't rely on more than Illuminate\Container\Container.

**Comment by crynobone** [comment]
<https://github.com/laravel/framework/blob/5.0/src/Illuminate/Database/Capsule/Manager.php#L61>

**Comment by martinbean** [comment]
@crynobone Then I’d argue why the docblock contained the `Application` type hint in the first place, but any managers I’ve found in Laravel’s source take an instance of `Illuminate\Foundation\Application`, as does my implementation, which is why I submitted this Pull Request.

**Pull request closed without merging**